### ISSUE TITLE
create plate maps in Markdown format

### ISSUE THREAD

**Issue by dave-doty** [DNA-and-Natural-Algorithms-Group/alhambra#11]
The method `Mix.table` creates a table that is useful for summarizing what to pipette, but it lists the plate wells as a comma-separated string, which is not easy to match visually to a 96-well or 384-well plate. 

Make a way to create plate maps in Markdown format that visually represent plate maps, e.g.,

```
## mix, 5.00 µl each
|     | 1    | 2   | 3   | 4   | 5   | 6   | 7   | 8   | 9   | 10   | 11   | 12   |
|-----|------|-----|-----|-----|-----|-----|-----|-----|-----|------|------|------|
| A   | mon0 |     |     |     |     |     |     |     |     |      |      |      |
| B   |      |     |     |     |     |     |     |     |     |      |      |      |
| C   |      |     |     |     |     |     |     |     |     |      |      |      |
| D   | mon3 |     |     |     |     |     |     |     |     |      |      |      |
| E   | mon4 |     |     |     |     |     |     |     |     |      |      |      |
| F   |      |     |     |     |     |     |     |     |     |      |      |      |
| G   |      |     |     |     |     |     |     |     |     |      |      |      |
| H   |      |     |     |     |     |     |     |     |     |      |      |      |
```
It should be customizable to put an arbitrary string in each well instead of the strand name:
```
## 5 monomer synthesis, 5.00 µl each
|     | 1   | 2   | 3   | 4   | 5   | 6   | 7   | 8   | 9   | 10   | 11   | 12   |
|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|------|------|------|
| A   | X   |     |     |     |     |     |     |     |     |      |      |      |
| B   |     |     |     |     |     |     |     |     |     |      |      |      |
| C   |     |     |     |     |     |     |     |     |     |      |      |      |
| D   | X   |     |     |     |     |     |     |     |     |      |      |      |
| E   | X   |     |     |     |     |     |     |     |     |      |      |      |
| F   |     |     |     |     |     |     |     |     |     |      |      |      |
| G   |     |     |     |     |     |     |     |     |     |      |      |      |
| H   |     |     |     |     |     |     |     |     |     |      |      |      |
```

**Comment by dave-doty** [comment]
# Release notes

## Create plate maps in Markdown format

A few methods have been added to the `Mix` class that allow exporting plate maps in a more visual way than by listing the well names in a comma-separated string (as `Mix.table()` does).

For example, the following code:

```python
import alhambra.mixes as am
strands = [
    am.Strand(name='strand 1', concentration="200 uM", plate='plate 1', well='A1', sequence='AAA'), 
    am.Strand(name='strand 2', concentration="200 uM", plate='plate 1', well='A2', sequence='CCC'),
    am.Strand(name='strand 3', concentration="100 uM", plate='plate 1', well='B3', sequence='GGG'),
    am.Strand(name='strand 4', concentration="100 uM", plate='plate 1', well='B4', sequence='GGG'),
    am.Strand(name='strand 5', concentration="100 uM", plate='plate 1', well='C5', sequence='GGG'),
    am.Strand(name='strand 6', concentration="100 uM", plate='plate 2', well='A1', sequence='GGG'),
    am.Strand(name='strand 7', concentration="100 uM", plate='plate 2', well='A2', sequence='GGG'),
    am.Strand(name='strand 8', concentration="100 uM", plate='plate 2', well='B1', sequence='GGG'),
    am.Strand(name='strand 9', concentration="100 uM", plate='plate 2', well='B2', sequence='GGG'),
]
mix = am.Mix(
    actions=[
        am.MultiFixedConcentration(components=strands, fixed_concentration="10 uM", set_name='mix')
    ],
    name="mix",
    fixed_total_volume="200 uL",
)
plate_maps = mix.plate_maps()
for plate_map in plate_maps:
    print(plate_map.to_markdown(well_marker='X') + '\n')
```

prints this:

```
## plate 1, 10.00 µl each
|     | 1   | 2   | 3   | 4   | 5   | 6   | 7   | 8   | 9   | 10   | 11   | 12   |
|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|------|------|------|
| A   | X   | X   |     |     |     |     |     |     |     |      |      |      |
| B   |     |     |     |     |     |     |     |     |     |      |      |      |
| C   |     |     |     |     |     |     |     |     |     |      |      |      |
| D   |     |     |     |     |     |     |     |     |     |      |      |      |
| E   |     |     |     |     |     |     |     |     |     |      |      |      |
| F   |     |     |     |     |     |     |     |     |     |      |      |      |
| G   |     |     |     |     |     |     |     |     |     |      |      |      |
| H   |     |     |     |     |     |     |     |     |     |      |      |      |

## plate 1, 20.0 µl each
|     | 1   | 2   | 3   | 4   | 5   | 6   | 7   | 8   | 9   | 10   | 11   | 12   |
|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|------|------|------|
| A   |     |     |     |     |     |     |     |     |     |      |      |      |
| B   |     |     | X   | X   |     |     |     |     |     |      |      |      |
| C   |     |     |     |     | X   |     |     |     |     |      |      |      |
| D   |     |     |     |     |     |     |     |     |     |      |      |      |
| E   |     |     |     |     |     |     |     |     |     |      |      |      |
| F   |     |     |     |     |     |     |     |     |     |      |      |      |
| G   |     |     |     |     |     |     |     |     |     |      |      |      |
| H   |     |     |     |     |     |     |     |     |     |      |      |      |

## plate 2, 20.0 µl each
|     | 1   | 2   | 3   | 4   | 5   | 6   | 7   | 8   | 9   | 10   | 11   | 12   |
|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|------|------|------|
| A   | X   | X   |     |     |     |     |     |     |     |      |      |      |
| B   | X   | X   |     |     |     |     |     |     |     |      |      |      |
| C   |     |     |     |     |     |     |     |     |     |      |      |      |
| D   |     |     |     |     |     |     |     |     |     |      |      |      |
| E   |     |     |     |     |     |     |     |     |     |      |      |      |
| F   |     |     |     |     |     |     |     |     |     |      |      |      |
| G   |     |     |     |     |     |     |     |     |     |      |      |      |
| H   |     |     |     |     |     |     |     |     |     |      |      |      |
```

In a jupyter notebook, the Markdown can be rendered nicely with `display` and `Markdown` from the package `IPython.display`:

```python
plate_maps = mix.plate_maps()
maps_strs = '\n\n'.join(plate_map.to_markdown(well_marker='X'))
from IPython.display import display, Markdown
display(Markdown(maps_strs))
```

which displays

## plate 1, 10.00 µl each
|     | 1   | 2   | 3   | 4   | 5   | 6   | 7   | 8   | 9   | 10   | 11   | 12   |
|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|------|------|------|
| A   | X   | X   |     |     |     |     |     |     |     |      |      |      |
| B   |     |     |     |     |     |     |     |     |     |      |      |      |
| C   |     |     |     |     |     |     |     |     |     |      |      |      |
| D   |     |     |     |     |     |     |     |     |     |      |      |      |
| E   |     |     |     |     |     |     |     |     |     |      |      |      |
| F   |     |     |     |     |     |     |     |     |     |      |      |      |
| G   |     |     |     |     |     |     |     |     |     |      |      |      |
| H   |     |     |     |     |     |     |     |     |     |      |      |      |

## plate 1, 20.0 µl each
|     | 1   | 2   | 3   | 4   | 5   | 6   | 7   | 8   | 9   | 10   | 11   | 12   |
|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|------|------|------|
| A   |     |     |     |     |     |     |     |     |     |      |      |      |
| B   |     |     | X   | X   |     |     |     |     |     |      |      |      |
| C   |     |     |     |     | X   |     |     |     |     |      |      |      |
| D   |     |     |     |     |     |     |     |     |     |      |      |      |
| E   |     |     |     |     |     |     |     |     |     |      |      |      |
| F   |     |     |     |     |     |     |     |     |     |      |      |      |
| G   |     |     |     |     |     |     |     |     |     |      |      |      |
| H   |     |     |     |     |     |     |     |     |     |      |      |      |

## plate 2, 20.0 µl each
|     | 1   | 2   | 3   | 4   | 5   | 6   | 7   | 8   | 9   | 10   | 11   | 12   |
|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|------|------|------|
| A   | X   | X   |     |     |     |     |     |     |     |      |      |      |
| B   | X   | X   |     |     |     |     |     |     |     |      |      |      |
| C   |     |     |     |     |     |     |     |     |     |      |      |      |
| D   |     |     |     |     |     |     |     |     |     |      |      |      |
| E   |     |     |     |     |     |     |     |     |     |      |      |      |
| F   |     |     |     |     |     |     |     |     |     |      |      |      |
| G   |     |     |     |     |     |     |     |     |     |      |      |      |
| H   |     |     |     |     |     |     |     |     |     |      |      |      |



By leaving off the parameter `well_marker` in `plate_map.to_markdown()`, the strand name is placed in the table:

```python
print(plate_map.to_markdown()
```

```
## plate 1, 20.0 µl each
|     | 1   | 2   | 3        | 4        | 5        | 6   | 7   | 8   | 9   | 10   | 11   | 12   |
|-----|-----|-----|----------|----------|----------|-----|-----|-----|-----|------|------|------|
| A   |     |     |          |          |          |     |     |     |     |      |      |      |
| B   |     |     | strand 3 | strand 4 |          |     |     |     |     |      |      |      |
| C   |     |     |          |          | strand 5 |     |     |     |     |      |      |      |
| D   |     |     |          |          |          |     |     |     |     |      |      |      |
| E   |     |     |          |          |          |     |     |     |     |      |      |      |
| F   |     |     |          |          |          |     |     |     |     |      |      |      |
| G   |     |     |          |          |          |     |     |     |     |      |      |      |
| H   |     |     |          |          |          |     |     |     |     |      |      |      |
```

The `PlateMap` objects in the list returned by the method `Mix.plate_maps()` has the data in a structured format in case you want to use it to create your own representation of a plate map. The following are its fields:

```python
plate_name: str # name of the plate
plate_type: scadnano.PlateType # type of the plate: 96-well or 384-well; this is a scadnano class: https://scadnano-python-package.readthedocs.io/en/latest/#scadnano.PlateType
vol_each: Quantity[Decimal] # volume of each well to pipette (is same for every well in the plate map)
well_to_strand_name: dict[str, str] # dictionary mapping each well (represented as a string such as "A3") to the name of the strand in that well
```

Note that any strands in the `Mix` coming from test tubes (instead of plates) are *ignored*. To get a representation of those strands in the mix, call `Mix.tubes_markdown()`. For example:

```python
import alhambra.mixes as am
strands = [
    am.Strand(name='strand 1', concentration="200 uM", plate='tube', sequence='AAA'), 
    am.Strand(name='strand 2', concentration="200 uM", plate='tube', sequence='CCC'),
    am.Strand(name='strand 3', concentration="100 uM", plate='tube', sequence='GGG'),
]
mix = am.Mix(
    actions=[
        am.MultiFixedConcentration(components=strands, fixed_concentration="10 uM", set_name='mix')
    ],
    name="mix",
    fixed_total_volume="200 uL",
)
print(mix.tubes_markdown())
```

prints

```
## tubes, 10.00 µl each
strand 1
strand 2
## tubes, 20.0 µl each
strand 3
```

More structured data with the same information is available by calling `Mix.vol_to_tube_names`, which returns a dict mapping a volume (represented as `Quantity[Decimal]`) to a list of strands in tubes that should be pipetted with that volume:

```python
from pprint import pprint
pprint(mix.vol_to_tube_names())
```

```
{<Quantity(10.00, 'microliter')>: ['strand 1', 'strand 2'],
 <Quantity(20.0, 'microliter')>: ['strand 3']}
```

**Comment by dave-doty** [comment]
Closed in #12
